File: fizzbuzz/fizzbuzz.c
/* FizzBuzz, Andrew Cashner, 2017/03/12 
 * Print numbers 1 to 100; if divisible by 3, print "Fizz" instead; if by 5,
 * "Buzz", if by 15, "FizzBuzz".
 */

#include <stdio.h>

int main(void) {
    int i;
    for (i = 1; i <= 100; ++i) {
        if (i % 15 == 0) {
            printf("FizzBuzz ");
        } else if (i % 3 == 0) {
            printf("Fizz ");
        } else if (i % 5 == 0) {
            printf("Buzz ");
        } else {
            printf("%d ", i);
        }
    }
    printf("\n");
    return(0);
}

File: fizzbuzz/fizzbuzz.rb
#!/usr/bin/env ruby

# Fizz Buzz, Andrew Cashner, 2017/04/17

class FizzBuzz
    def initialize(min, max)
        @min = min
        @max = max
    end
    def test
        if (@num % 15 == 0) 
            @msg = "FizzBuzz"
        elsif (@num % 3 == 0)
            @msg = "Fizz"
        elsif (@num % 5 == 0)
            @msg = "Buzz"
        else 
            @msg = @num
        end
        print "#{@msg} "
    end
    def iterate
        @num = @min
        while @num < @max
            self.test
            @num += 1
        end
        print "\n"
    end
end

fizzbuzz = FizzBuzz.new(1, 100)
fizzbuzz.iterate
